src/main/webapp/scripts/appscript.js

```diff
@@ -17,25 +17,52 @@
   * Customer ID dropdown.
   */
 async function populateCustomerList() {
-  // TODO: Implement functionality for hashmap parsing instead of simple array.
+  // Fetches json dictionary from VendorServlet.
   var response = await fetch('/VendorServlet');
 
-  // Response is a json array that is parsed into json.
-  var customerJson = await response.json();
-
-  console.log(customerJson);
+  var vendorJson = await response.json();
 
   // Loops over this array to make the options for the Customer ID dropdown.
   var vendorHtml = '';
-  for (var i = 0; i < customerJson.length; i++) {
-    html += '<option value="' + customerJson[i] + '">'
-    + customerJson[i] + '</option>';
+  for (var key in vendorJson) {
+    vendorHtml += '<option value="' + key + '">'
+    + key + '</option>';
   }
 
   // Adds the options to the page, allowing them to be viewed. 
   document.getElementById('customer-ids').innerHTML = vendorHtml;
 }
 
+/**
+ * Fetch account IDs from VendorServlet and add them to the Account ID
+ * dropdown.
+ */
+async function populateAccountList() {
+  // Fetches json dictionary from VendorServlet.
+  var response = await fetch('/VendorServlet');
+
+  var vendorJson = await response.json();
+
+  // Takes the value of the Vendor ID <select> element.
+  var vendorValue = document.getElementById('customer-ids').value;
+
+  // Loops over the dictionary to find the corresponding value in the dictionary
+  // and adds them to the Account ID <select> element.
+  var accountHtml = '';
+  for (var key in vendorJson) {
```

Comment:
no need to loop through the keys here. You can remove the loop and just say
`var array = vendorJson[vendorValue]`.

Consider renaming "array" to indicate what it actually holds